apache2/apache2_config.c

```diff
@@ -3698,6 +3702,34 @@ static const char *cmd_cache_transformations(cmd_parms *cmd, void *_dcfg,
     return NULL;
 }
 
+/**
+* \brief Add SecParseXMLIntoArgs configuration option
+*
+* \param cmd Pointer to configuration data
+* \param _dcfg Pointer to directory configuration
+* \param p1 Pointer to configuration option
+*
+* \retval NULL On failure
+* \retval apr_psprintf On Success
```

Comment:
I don't think this is correct. In one error case you return `NULL`, in the other `apr_psprintf`. In the second case `NULL` is the success value, if I'm not mistaken.

Reply:
You're right. I assume you think about [this](https://github.com/airween/ModSecurity/blob/v2/xmlargsfeat/apache2/apache2_config.c#L3720-L3724) condition (with its comment). Please check that comment: `// Normally useless code, left to be safe for the moment`. It exists in all directive handle fn, I didn't want to remove that.

So yes, you're right, but if once we will remove those conditions, then the comment will be true.

Reply:
No, even if you remove it, the function will return `NULL` on _success_, and `apr_psprintf` on _error_, which is the opposite of what the comment says.

Reply:
Ah, now I see - sorry and thank you. Fixed in c11bd6c.